backend/routes/user.js

```diff
@@ -0,0 +1,304 @@
+      where: { id: parseInt(id) },
+    });
+
+    if (!user) {
+      return res.status(404).json({ error: "User not found " });
+    }
+
+    res.json(user);
+  } catch (error) {
+    res.status(404).send("ID is not valid");
+  }
+});
+
+// update a user
+
+user.put("/:id", async (req, res) => {
+  const { id } = req.params;
+  const { id, email, password } = req.body;
+
+  try {
+    const existingUser = await prisma.user.findUnique({
+      where: { id: parseInt(id) },
+    });
+
+    if (!existingUser) {
+      return res.status(404).json({ error: "User not found" });
+    }
+
+    const updatedUser = await prisma.user.update({
+      where: { id: parseInt(id) },
+      data: {
+        id,
+        email,
+        password,
+      },
+    });
+
+    res.json(updatedUser);
+  } catch (error) {
+    res.status(400).json({ error: "Failed to update user" });
+  }
+});
+
+// add route to partially update user here
+
+
+// add or update user salary
+
+user.patch("/salary", async (req, res) => {
+  const { userId, salary } = req.body;
+
+  try {
+    const updatedSalary = await prisma.userSensitiveData.upsert({
+      where: { userId },
+      update: { salary },
+      create: {
+        userId,
+        salary,
+      },
+    });
+
+    res.status(200).json(updatedSalary);
+  } catch (error) {
+    res
+      .status(400)
+      .json({ error: "Failed to update salary", details: error.message });
+  }
+});
+
+// route to get a user's salary
+
+user.get("/salary", async (req, res) => {
+  const { userId, salary } = req.body;
+
+  try {
+    const userSalary = await prisma.userSensitiveData.findUnique({
+      where: { id: parseInt(id) },
+    });
+    if (!userSensitiveData) {
+      return res.status(404).json({ error: "Salary not found " });
+    }
+    res.json(salary);
+  } catch (error) {
+    res.status(404).send("ID is not valid");
+  }
+});
+
+// route to create financial data
+
+user.post("/financial-data", async (req, res) => {
+  const { userId, category, amount, date } = req.body;
+  if ((!userId || !category, amount == null || !date)) {
+    return res.status(400).json({ error: "Missing required fields!" });
+  }
+
+  try {
+    const newFinancialData = await prisma.FinancialData.create({
+      data: {
+        userId,
+        category,
+        amount: parseFloat(amount),
+        date: new Date(date),
+      },
+    });
+
+    res.status(201).json(newFinancialData);
+  } catch (error) {
+    res.status(500).json({ error: "Internal error" });
+  }
+});
+
+// route to get financial data
+
+user.get("/financial-data", async (req, res) => {
+  const { id } = req.params;
+  const { userId, category, amount, date } = req.body;
+
+  try {
+    const userFinancialData = await prisma.FinancialData.findUnique({
+      where: { id: parseInt(id) },
+    });
+    if (!userFinancialData) {
+      return res.status(404).json({ error: "Financial Data not found " });
+    }
+    res.json(category, amount, date);
```

Comment:
consider bundling them in an object instead of passing as separate args
```suggestion
    res.json({category, amount, date});
```